TechnicalDocuments/CompomentComposition.md

```diff
@@ -0,0 +1,63 @@
+# Bento Component Composition Using `UIStackView`
+
+#### Updated at: 23-3-2019
+
+## The scope of this document 
+Component Composition with `Bento` is a very wide topic due to its flexible design.
+`Bento` components can be composed in many ways.
+
+The purpose of this document is to provide documentation for __composing Bento components using `UIStackView`__.
+Other ways of composition are out of the scope of this document.
+
+## How to use component composition in your framework
+
+Well, the answer to this question is that your can't.
+The reasoning for this decision is that
+* we must reuse components via the `DesignLibrary`
+* If you need something custom and customizable then using `UIStackView`s probably won't be good enough for you
+* We need to be flexible and to depend on abstractions.
+
+#### Where and how to use component composition
+In `BabylonUI` there is a new namespace called `UI` which is the place for `Atomic Components`.
+
+- `Atomic Components` are the `Bento` Components version of `UIKit`'s components so the nomenclature is similar  like `UILabel` and `UI.Label`, `UIButton` and `UI.Button` etc 
+- `Atomic Components` use `Stylesheets` for styling
+- `Atomic Components` should behave like their `UIKit` counterpart.
+
+## Code Sample
+
+```swift
+[
+    UI.Label(
+      text: "Some Title",
+      styleSheet: LabelStyleSheet()
+          .compose(\.textColor, tokens.colors.black)
+          .compose(\.font, font(fontAttributes: tokens.typography.body))
+    ).asAnyRenderable(),
+    UI.Label(
+        text: "Some Details",
+        styleSheet: LabelStyleSheet()
+            .compose(\.textColor, tokens.colors.grey700)
+            .compose(\.font, font(fontAttributes: tokens.typography.subhead))
+    ).asAnyRenderable()
+]
+.stack(
+    axis: .vertical,
+    spacing: 2
+)
+```
+
+Currently `Renderable` provides some util methods like `card(..)`, `stack(...)`, `width(...)` and `height(...)`.
```

Comment:
Would be nice to mention that someone can make their own extensions. For example I just needed `minHeight` behaviour. I guess that we'll provide documentation how to create them later